js/hotgraphicPopupView.js

```diff
@@ -0,0 +1,141 @@
+define([
+    'core/js/adapt',
+    'core/js/views/diffView'
+], function(Adapt, DiffView) {
+
+    var HotGraphicPopup = DiffView.extend({
+
+        renderAttributes: [
+            "_hidePagination",
+            "_currentPosition",
+            "_currentIndex"
+        ],
+
+        className: function() {
+            return this.normalizeClassNames([
+                "hotgraphic-popup",
+                this.state ? this.state.get("_popupClasses") : ""
+            ]);
+        },
+
+        events: {
+            'click .hotgraphic-popup-done': 'closeHotGraphic',
+            'click .hotgraphic-popup-nav .back': 'previousHotGraphic',
+            'click .hotgraphic-popup-nav .next': 'nextHotGraphic'
+        },
+
+        postInitialize: function() {
+            var currentIndex = this.state.get("_currentIndex");
+            this.moveToIndex(currentIndex);
+        },
+
+        moveToIndex: function(index, initial) {
+            this.setVisited(index);
+
+            this.state.set({
+                "_currentIndex": index,
+                "_currentPosition": index+1,
+                "_total": this.model.getItemsCount()
+            });
+
+            this.setNavigationForItem(index);
+
+            // Make sure the focus is set correctly after the popup moves to a new index and is rerendered
+            this.listenToOnce(this, "postRender", function () {
+                this.$('.active').a11y_focus();
+            });
+        },
+
+        setVisited: function(index) {
+            var item = this.model.getItem(index);
+            item._isVisited = true;
+
+            $.a11y_alert("visited");
+        },
+
+        setNavigationForItem: function (index) {
+            var itemsCount = this.model.getItemsCount();
+
+            var allowCycle = this.model.get('_canCycleThroughPagination');
+            var isFirst = (index <= 0);
+            var isLast = (index >= itemsCount-1);
+            var isMiddle = (!isFirst && !isLast);
```

Comment:
Apologies I didn't query this at the time. Was `isMiddle` added for anything special?

Could just have `_showNext` logic as `allowCycle || !isLast` and `_showBack` as `allowCycle || !isFirst`?